### PR TITLE
Handle byte order mark in UTF-8 data

### PR DESCRIPTION
**Pull request by jkehres**
Skips the byte order mark when present at the start of the stream. See the following for details:
https://en.wikipedia.org/wiki/Byte_order_mark#UTF-8

### CODE REVIEW

File: test/data/test_utf8_bom.csv [hunk 1]
```diff
@@ -0,0 +1,3 @@
+﻿a,b,c
```
Comment (jkehres) [inline test/data/test_utf8_bom.csv]: Can't tell from the diff but the file contains a BOM at the start.

File: test/test.js [hunk 2]
```diff
@@ -1,12 +1,12 @@
 var test = require('tape')
 var fs = require('fs')
 var path = require('path')
-var eol = require('os').EOL
 var bops = require('bops')
 var spectrum = require('csv-spectrum')
 var concat = require('concat-stream')
 var csv = require('..')
 var read = fs.createReadStream
+var eol = '\n'
```
Comment (jkehres) [inline test/test.js]: Again a fix for Windows.

**Comment by shellscape** [comment]
@jkehres what are the conditions under which this happens?

**Comment by shellscape** [comment]
If this PR needs merging in the future, please rebase and ping, and we'll reopen it.

**Pull request closed without merging**